src/hiccup/compiler.clj

```diff
@@ -152,7 +156,7 @@
   "True if x is a literal value that can be rendered as-is."
   [x]
   (and (not (unevaluated? x))
-       (or (not (or (vector? x) (map? x)))
+       (or (not (or (vector? x) (map? x) (set? x)))
```

Comment:
Maybe I misunderstood the intention of this code. This escapes literals when they are _not_ treated specially (e.g. when you write `[:span {} {:foo "bar"}]` - not that it is too useful...). I thought it would be nice to treat sets the same way. But I revert this commit then.